File: library/src/main/java/com/ebr163/inspector/rule/pair/ConfirmPasswordRule.java
package com.ebr163.inspector.rule.pair;

import android.util.Pair;

import com.ebr163.inspector.rule.AbstractRule;

/**
 * Created by Bakht
 * on 10.01.2018.
 */

public class ConfirmPasswordRule extends AbstractRule<Pair<String, String>> {

    protected ConfirmPasswordRule(String errorMessage) {
        super(errorMessage);
    }

    @Override
    public boolean verify(Pair<String, String> value) {
        return value.first.equals(value.second);
    }
}

File: library/src/main/java/com/ebr163/inspector/rule/pair/ConfirmPasswordRule.kt
package com.ebr163.inspector.rule.pair

import android.util.Pair

import com.ebr163.inspector.rule.AbstractRule

/**
 * Created by Bakht
 * on 10.01.2018.
 */

class ConfirmPasswordRule(errorMessage: String) : AbstractRule<Pair<String, String>>(errorMessage) {

    override fun verify(value: Pair<String, String>?): Boolean = value?.first == value?.second
}
